packages/@cdktf/provider-generator/lib/get/generator/constants/provider-namespaces.ts

```diff
@@ -0,0 +1,614 @@
+  },
+  ssoadmin: {
+    name: "SSOAdmin",
+    comment: "AWS SSO Admin",
+  },
+  swf: {
+    name: "SWF",
+    comment: "AWS Simple Workflow Service",
+  },
+  sagemaker: {
+    name: "SageMaker",
+    comment: "AWS SageMaker",
+  },
+  secretsmanager: {
+    name: "SecretsManager",
+    comment: "AWS Secrets Manager",
+  },
+  securityhub: {
+    name: "SecurityHub",
+    comment: "AWS Security Hub",
+  },
+  serverlessapplicationrepository: {
+    name: "ServerlessApplicationRepository",
+    comment: "AWS Serverless Application Repository",
+  },
+  servicecatalog: {
+    name: "ServiceCatalog",
+    comment: "AWS Service Catalog",
+  },
+  service_discovery: {
+    name: "ServiceDiscovery",
+    comment: "AWS Service Discovery",
+  },
+  servicequotas: {
+    name: "ServiceQuotas",
+    comment: "AWS Service Quotas",
+  },
+  shield: {
+    name: "Shield",
+    comment: "AWS Shield Protection",
+  },
+  signer: {
+    name: "Signer",
+    comment: "AWS Signer",
+  },
+  simpledb: {
+    name: "SimpleDB",
+    comment: "AWS SimpleDB",
+  },
+  sfn: {
+    name: "SFN",
+    comment: "AWS Step Functions",
+  },
+  storagegateway: {
+    name: "StorageGateway",
+    comment: "AWS Storage Gateway",
+  },
+  synthetics: {
+    name: "Synthetics",
+    comment: "AWS Synthetics",
+  },
+  timestreamwrite: {
+    name: "TimestreamWrite",
+    comment: "AWS Timestream Write",
+  },
+  transfer: {
+    name: "Transfer",
+    comment: "AWS Transfer",
+  },
+  vpc: {
+    name: "VPC",
+    comment: "AWS VPC",
+    additionalPrefix: [
+      "customer_gateway",
+      "default_network_acl",
+      "default_route_table",
+      "default_security_group",
+      "default_subnet",
+      "default_vpc",
+      "egress_only_internet_gateway",
+      "flow_log",
+      "internet_gateway",
+      "main_route",
+      "nat_gateway",
+      "network",
+      "route_table",
+      "route",
+      "security_group",
+      "subnet",
+      "vpn",
+    ],
+  },
+  wafv2: {
+    name: "WAFV2",
+    comment: "AWS WAF V2",
+  },
+  wafregional: {
+    name: "WAFRegional",
+    comment: "AWS WAF Regional",
+  },
+  waf: {
+    name: "WAF",
+    comment: "AWS WAF",
+  },
+  worklink: {
+    name: "WorkLink",
+    comment: "AWS WorkLink",
+  },
+  workspaces: {
+    name: "Workspaces",
+    comment: "AWS WorkSpaces",
+  },
+  xray: {
+    name: "XRay",
+    comment: "AWS X-Ray",
+  },
+  _datasources: {
+    name: "DataSources",
+    comment: "AWS Data Sources",
+    additionalPrefix: [
+      "arn",
+      "availability_zone",
+      "billing_service_account",
+      "caller_identity",
+      "canonical_user_id",
+      "elb_hosted_zone_id",
+      "elb",
+      "ip_ranges",
+      "launch_configuration",
+      "partition",
+      "prefix_list",
+      "region",
+    ],
+  },
+};
+
+const PROVIDER_NAMESPACES: Record<string, Record<Prefix, ResourceNamespace>> = {
+  aws,
+};
+
+// Comments per namespace mapping
+export function getResourceNamespace(
+  provider: string,
```

Comment:
Will provider already be "normalized" here? Reason: Passing "hashicorp/aws" should be treated as "aws" (which could happen when a provider is specified as `hashicorp/aws` in the `cdktf.json` file.

Reply:
Yes, it's normalized: https://github.com/hashicorp/terraform-cdk/blob/5bf105cb1a8cf3477722c93d8869d431d75f7888/packages/@cdktf/provider-generator/lib/get/generator/provider-generator.ts#L74